packages/web_ui/src/util/websocket.jsx

```diff
@@ -119,6 +123,26 @@ export class Control extends libLink.Link {
 		console.log("WS", message.data.direction, message.data.content);
 	}
 
+	async onLiveSlaveAdded(id, handler) {
+		if (this.liveUpdateSlaveHandlers.size === 0) {
+			libLink.messages.setLiveSlaveSubscription.send(this, { connect: true });
+		}
+		this.liveUpdateSlaveHandlers.set(id, handler);
+	}
+
+	async offLiveSlaveAdded(id) {
+		this.liveUpdateSlaveHandlers.delete(id);
+		if (this.liveUpdateSlaveHandlers.size === 0) {
+			libLink.messages.setLiveSlaveSubscription.send(this, { connect: false });
+		}
+	}
```

Comment:
Why do you identify the handler with an arbitrary `id`? This is different from how the onInstanceLog and offInstanceLog works and makes it more difficult to make components properly add and remove their handlers from this event.

Also logic is missing to re-establish the slaveUpdateSubscription after a reconnect event like there is for the log handlers (see the connect event handler in the constructor).